rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -125,27 +128,32 @@ class ClocksState final
   {
     std::lock_guard<std::mutex> clock_guard(clock->get_clock_mutex());
 
-    // Do change
-    if (!set_ros_time_enabled && clock->ros_time_is_active()) {
-      auto ret = rcl_disable_ros_time_override(clock->get_clock_handle());
-      if (ret != RCL_RET_OK) {
-        rclcpp::exceptions::throw_from_rcl_error(
-          ret, "Failed to disable ros_time_override_status");
+    if (clock->get_clock_type() == RCL_ROS_TIME) {
+      // Do change
+      if (!set_ros_time_enabled && clock->ros_time_is_active()) {
+        auto ret = rcl_disable_ros_time_override(clock->get_clock_handle());
+        if (ret != RCL_RET_OK) {
+          rclcpp::exceptions::throw_from_rcl_error(
+            ret, "Failed to disable ros_time_override_status");
+        }
+      } else if (set_ros_time_enabled && !clock->ros_time_is_active()) {
```

Comment:
minor: I think that the code is more readable if rather than having this `} else if (set_ros_time_enabled && !clock->ros_time_is_active()) {` and then `} else if (set_ros_time_enabled) {` we rather do

```
} else if (set_ros_time_enabled) {
  if (clock->ros_time_is_active()) {
    throw std::invalid_argument(....)
  }
  ....
}
```

Reply:
After thinking about this, I still think it's easier to maintain if we keep the logic for RCL_ROS_TIME clock in one block(from line 131 to 152), and the rest for other clock types in another block (from 154 to 156). What do you think, @alsora?

Reply:
Good for me, this was just a minor comment